ts-client/src/example.ts

```diff
@@ -109,38 +199,136 @@ async function getPositionsState(dlmmPool: DLMM) {
+  } catch (error) {
+    console.log("🚀 ~ error:", JSON.parse(JSON.stringify(error)));
+  }
+}
+
+async function removeBalancePositionLiquidity(dlmmPool: DLMM) {
+  const userPosition = userPositions.find(({ publicKey }) =>
+    publicKey.equals(newBalancePosition.publicKey)
+  );
+  // Remove Liquidity
+  const binIdsToRemove = userPosition.positionData.positionBinData.map(
+    (bin) => bin.binId
+  );
+  const removeLiquidityTx = await dlmmPool.removeLiquidity({
+    position: userPosition.publicKey,
+    user: user.publicKey,
+    binIds: binIdsToRemove,
+    liquiditiesBpsToRemove: new Array(binIdsToRemove.length).fill(
+      new BN(100 * 100)
+    ), // 100% (range from 0 to 100)
+    shouldClaimAndClose: true, // should claim swap fee and close position together
+  });
+
+  try {
+    for (let tx of Array.isArray(removeLiquidityTx)
+      ? removeLiquidityTx
+      : [removeLiquidityTx]) {
+      const removeBalanceLiquidityTxHash = await sendAndConfirmTransaction(
+        connection,
+        tx,
+        [user],
+        { skipPreflight: false, preflightCommitment: "singleGossip" }
+      );
+      console.log(
+        "🚀 ~ removeBalanceLiquidityTxHash:",
+        removeBalanceLiquidityTxHash
+      );
+    }
+  } catch (error) {
+    console.log("🚀 ~ error:", JSON.parse(JSON.stringify(error)));
+  }
+}
+
+async function removeImbalancePositionLiquidity(dlmmPool: DLMM) {
+  const userPosition = userPositions.find(({ publicKey }) =>
+    publicKey.equals(newImbalancePosition.publicKey)
+  );
+  // Remove Liquidity
+  const binIdsToRemove = userPosition.positionData.positionBinData.map(
+    (bin) => bin.binId
+  );
+  const removeLiquidityTx = await dlmmPool.removeLiquidity({
+    position: userPosition.publicKey,
+    user: user.publicKey,
+    binIds: binIdsToRemove,
+    liquiditiesBpsToRemove: new Array(binIdsToRemove.length).fill(
+      new BN(100 * 100)
+    ), // 100% (range from 0 to 100)
+    shouldClaimAndClose: true, // should claim swap fee and close position together
   });
 
   try {
-    for (let tx of Array.isArray(addLiquidityTx)
-      ? addLiquidityTx
-      : [addLiquidityTx]) {
-      const addLiquidityTxHash = await sendAndConfirmTransaction(
+    for (let tx of Array.isArray(removeLiquidityTx)
+      ? removeLiquidityTx
+      : [removeLiquidityTx]) {
+      const removeImbalanceLiquidityTxHash = await sendAndConfirmTransaction(
         connection,
         tx,
-        [user, newPosition]
+        [user],
+        { skipPreflight: false, preflightCommitment: "singleGossip" }
+      );
+      console.log(
+        "🚀 ~ removeImbalanceLiquidityTxHash:",
+        removeImbalanceLiquidityTxHash
       );
-      console.log("🚀 ~ addLiquidityTxHash:", addLiquidityTxHash);
     }
   } catch (error) {
     console.log("🚀 ~ error:", JSON.parse(JSON.stringify(error)));
   }
 }
 
-async function removeLiquidity(dlmmPool: DLMM) {
+async function removeOneSidePositionLiquidity(dlmmPool: DLMM) {
```

Comment:
^Ditto, unless you add logic to withdraw only left / right side of the positions only by using pool state active bin ID as the median.